monkey/infection_monkey/master/automated_master.py

```diff
@@ -55,6 +60,9 @@ def __init__(
         self._master_thread = create_daemon_thread(
             target=self._run_master_thread, name="AutomatedMasterThread"
         )
+        self._heartbeat_thread = create_daemon_thread(
+            target=self._run_heartbeat_thread, name="HeartbeatThread"
+        )
```

Comment:
This is not a function of the master, it's a function of the agent. This needs to be moved to `monkey.py`

I'd recommend a class simply called `Heart()` that has a `start()` and `stop()` method. Build in `monkey.py` and `start()` the heartbeat after agent registration. `stop()` the heartbeat just before closing outbound tunnel.